packages/web-components/@sfx/products/test/index.test.ts

```diff
@@ -0,0 +1,95 @@
+import { expect, sinon, spy, stub } from './utils';
+import { Products } from '../src/index';
+
+describe('Products Component ', () => {
+  let component: any = {};
+  beforeEach(() => {
+    component = new Products();
+  });
+
+  describe('general', () => {
+    it('should default to have an empty array of products', () => {
+      expect(component.products).to.deep.equal([]);
+    });
+
+    it('should default to 12 maxItems', () => {
+      expect(component.maxItems).to.deep.equal(12);
+    });
+  });
+
+  describe('properties', () => {
+    it('should reflect maxItems as a property', () => {
+      expect(Products.properties.maxItems.reflect).to.equal(true);
+    });
+  });
+
+  describe('connectedCallback', () => {
+    it('should set up event listeners after connecting', () => {
+      const listenStub = sinon.stub(component, 'setUpEventListeners');
+
+      component.connectedCallback();
+
+      expect(listenStub).to.be.called;
+    });
+  });
+
+  describe('setProductsFromEvent', () => {
+    it('should set the event products payload into the component', () => {
+      const products = [1, 2, 3];
+      const event = { detail: { products }};
+      expect(component.products).to.deep.equal([]);
```

Comment:
Is this assertion necessary? I would either set it explicitly to an empty array or assume that it's not going to be `products`.